### PR TITLE
Add some documentation for kuttl tests

### PR DESCRIPTION
**Pull request by cescgina**
Add some basic documentation for writing kuttl tests. The goal is to
capture some common pitfalls/tricks when using kuttl and make clear
which conventions have been used to achieve consistency.

### CODE REVIEW

File: kuttl_tests.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Kuttl is a tool for testing Kubernetes operators. Each kuttl test suite
+consists of a series of test steps that might modify the state of the cluster
+or check its state. A kuttl test step will use all yaml files that begin with
+its numeric index. For example, if we had a file 00-deploy.yaml and
+a 00-assert.yaml, kuttl would apply the changes described by the file
+00-deploy.yaml and then wait until the state of the cluster matches the one
+found in 00-assert.yaml. If it's found, then the step is successful. If after
+the timeout, the state of the cluster does not match the assert, then the test
+fails. Kuttl also supports errors files, which will check that the cluster does
+not match the state described in the file.
```
Comment (marios) [inline kuttl_tests.md:13]: (i.e. the inverse of 'assert') 

just suggestion

File: kuttl_tests.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+What should be tested for in a kuttl suite depends on the details of the
+operator itself. A good place to find more information is the controller.go
+file for the operator, there you should find
+hints on what resources does the operator create and/or manage. Another source
+of information would be to contact the people listed as approvers in the
+operator's `OWNERS` file. Below are some example of things to assert while
+writing the tests (they might not apply to all operators):
```
Comment (marios) [inline kuttl_tests.md:21]: i would also suggest mentioning the operator API as a source of info for which tests we can write eg https://github.com/openstack-k8s-operators/keystone-operator/blob/master/api/bases/keystone.openstack.org_keystoneapis.yaml to see what attributes the operator exposes and things like configOverwrite for the keystone example

**Comment by marios** [comment]
lgtm thanks for adding

File: kuttl_tests.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Below are some conventions that are assumed for kuttl tests in
+openstack-k8s-operators:
+
+* There should be a kuttl-test.yaml file at the root of the operator
+repo which contains the kuttl configuration.
+* There should be `${OPERATOR_NAME}_kuttl` and `${OPERATOR_NAME}_kuttl_run` targets in `install_yamls`. The first will install all dependencies, run the  tests and cleanup, while the latter will simply run the tests, without further modifying the environment.
+* There should be an assert file under
+`operator-path/tests/kuttl/common/assert_sample_deployment.yaml`.
+This assert file should be used to validate the deployment
+of the operator. In parallel, a target called `${OPERATOR_NAME}\_deploy_validate`
```
Comment (abays) [inline kuttl_tests.md:41]: Should `${OPERATOR_NAME}\_deploy_validate` be `${OPERATOR_NAME}_deploy_validate` instead (i.e. remove the `\`)?

File: kuttl_tests.md (hunk 1 above)
Comment (cescgina) [reply kuttl_tests.md]: Added.

File: kuttl_tests.md (hunk 2 above)
Comment (cescgina) [reply kuttl_tests.md]: Good point, I've added it.

File: kuttl_tests.md (hunk 3 above)
Comment (cescgina) [reply kuttl_tests.md]: Fixed, thanks!

**Pull request merged**